src/modules/User/components/Payment/Payment.tsx

```diff
@@ -1,83 +1,123 @@
+import { Box, Grid, Typography } from "@mui/material"; // Fixed the Grid import
+import { FormEvent } from "react";
+import { Base_Url } from "../../../../constants/End_Points";
+import PaymentIcon from "@mui/icons-material/Payment";
+import axios from "axios";
+import { loadStripe } from "@stripe/stripe-js";
 import { toast } from "react-toastify";
+import { useNavigate, useParams } from "react-router-dom";
+
+// Define the type for the params to be passed to the CheckoutForm
+interface CheckoutFormProps {
+  bookingId: string | undefined;
+}
+
+const stripePromise = loadStripe(
+  "pk_test_51OTjURBQWp069pqTmqhKZHNNd3kMf9TTynJtLJQIJDOSYcGM7xz3DabzCzE7bTxvuYMY0IX96OHBjsysHEKIrwCK006Mu7mKw8"
+);
 
-const stripe = loadStripe("pk_test_51OTjURBQWp069pqTmqhKZHNNd3kMf9TTynJtLJQIJDOSYcGM7xz3DabzCzE7bTxvuYMY0IX96OHBjsysHEKIrwCK006Mu7mKw8")
 export default function Payment() {
-    return <Elements stripe={stripe}>
+  const { bookingId } = useParams();
 
-    <CheckoutForm/>
-  </Elements>
+  return (
+    <Elements stripe={stripePromise}>
+      <CheckoutForm bookingId={bookingId} />
+    </Elements>
+  );
 }
 
-const CheckoutForm =()=>{
-
-    const stripe = useStripe ()
-    const elements =useElements()
-    const handelSubmit = async(e:FormEvent)=>{
-        e.preventDefault()
-        if (!elements || !stripe ) return;
-         const cardElement =elements?.getElement("card")
-         const addressElement=elements.getElement("address")
-        if (!cardElement  || !addressElement) return;
-        const result = await stripe?.createToken (cardElement)
-        const addressValue = await addressElement.getValue()
-       console.log(addressValue)
-  if (result.error) {
-    console.log({error :result.error})
-    return;
-  }
-  await payBooking(result.token.id)
-  console.log(result.token.id)
-    }
+// Make sure that bookingId is properly passed as a prop
+const CheckoutForm = ({ bookingId }: CheckoutFormProps) => {
+  const navigate = useNavigate();
+  const stripe = useStripe();
+  const elements = useElements();
 
-    return <>
-    <Grid2 container   >
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    if (!elements || !stripe) return;
 
+    const cardElement = elements.getElement(CardElement);
+    const addressElement = elements.getElement(AddressElement);
 
+    if (!cardElement || !addressElement) return;
+
+    const result = await stripe.createToken(cardElement);
+    if (result.error) {
+      toast.error(result.error.message);
+      return;
+    }
 
-    <form className="form-wrapper" onSubmit={handelSubmit} >
-          
-  <Typography  sx={{color:"#152C5B" , fontFamily:"Poppins",fontSize:"28px",textAlign:"center"}}>Payment
-  <PaymentIcon fontSize="large" sx={{color:"#FF498B" }} className="PaymentIcon" />
-  </Typography>
+    // Ensure bookingId is passed correctly
+    await payBooking(result.token.id, bookingId, navigate);
+  };
+
+  return (
+    <Grid container>
```

Comment:
We can display the Grid as a form and provide the props given to the form to it. This will help us remove unnecessary markup in our component.